plugins/rts54xx/fu-plugin-rts54xx.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin.h"
+#include "fu-plugin-vfuncs.h"
+
+#include "fu-rts54xx-device.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_REQUIRES_QUIRK, FU_QUIRKS_PLUGIN);
+}
+
+gboolean
+fu_plugin_update (FuPlugin *plugin,
+		  FuDevice *device,
+		  GBytes *blob_fw,
+		  FwupdInstallFlags flags,
+		  GError **error)
+{
+	g_autoptr(FuDeviceLocker) locker = NULL;
+
+	/* check */
+	if (g_bytes_get_size (blob_fw) % 0x10000 != 0) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INVALID_FILE,
+			     "File not expected size, expected multiple of 0x%04x",
+			     (guint) 0x10000);
+		return FALSE;
+	}
```

Comment:
Isn't the bank size fixed also?  I would think you can check for that here too.

Reply:
Do know how to get the bank size?

Reply:
I don't know how to get it programmatically from the hub.  I'd check with Realtek on this.  It can always be put into quirks if it's variable and not queryable, but I think it's a good idea to test for this.  Otherwise you're going to need to trust that the writing to an empty bank doesn't overflow, corrupt, etc.